File: 20220810/모범답안/20220810_03.js/inputK.js
function solution(N, K){
    var arr = N.toString().split("");
    var insert = 0;
  
    if(N >= 0){
      for(var i = 0; i < arr.length; i++){
        if(Number(arr[i]) <= K){
          arr.splice(i, 0, K.toString());
          insert = 1;
          break;
        }
      }
    } else{
      for(var i = 1; i < arr.length; i++){
        if(Number(arr[i]) > K){
          arr.splice(i, 0, K.toString());
          insert = 1;
          break;
        }
      }
  
    }
  
    if(insert == 0){
      arr.push(K.toString());
    }
  
    return Number(arr.join(""));
  }

File: 20220810/모범답안/20220810_03.js/inputK.py
def solution(N, K):
    arr = [ s for s in str(N) ];
    insert = 0

    if N >= 0:
        for i in range(0, len(arr)):
            if int(arr[i]) <= K:
                arr.insert(i, str(K))
                insert = 1
                break
    else:
        for i in range(1, len(arr)):
            if int(arr[i]) > K:
                arr.insert(i, str(K))
                insert = 1
                break

    if insert == 0:
        arr.append(str(K))

    return (int("".join(arr)))
